extensions/wikia/AttributeServiceMWApi/AttributeServiceMWApiController.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+class AttributeServiceMWApiController extends WikiaController {
+
+    /**
+     * Helper function for the attribute service to clear user MediaWiki.
+     * This is called when a client other than MW updates an attribute
+     * using the attribute service to clear the User cache.
+     */
+    public function purgeUserCache() {
+
+        $userId = $this->request->getVal( 'userId' );
+        $signature = $this->request->getVal( 'signature' );
+
+       if ( !$this->assertValidParameters( $userId, $signature ) ) {
+           return;
+       }
+
+        if ( !$this->assertValidSignature( $userId, $signature ) ) {
+            return;
+        }
+
+        User::clearUserCache( $userId );
```

Comment:
Hmm, I was trying `User::invalidateCache` at first but it wasn't working properly. It looks like it loads the previous values form cache, deletes the cache, and then sets that old cache again...I do need to cleared the shared cache though, and update, the `user_touched`. Calling `User::clearUserCache( $userId );` first seems to fix it.